02-cpu-configure/cpu-configure.sh

```diff
@@ -2,108 +2,100 @@
-
-BPM_EPICS_CFG_FILE="/etc/sysconfig/bpm-epics-ioc"
-BPM_HALCS_CFG_FILE="/usr/local/etc/halcs/halcs.cfg"
-TIM_RX_EPICS_CFG_FILE="/etc/sysconfig/tim-rx-epics-ioc"
-
-## Ask sudo password only once and
-## keep updating sudo timestamp to
-## avoid asking again
-#sudo -v
-#while true; do sudo -n true; sleep 60; kill -0 "$$" || \
-#    exit; done 2>/dev/null &
-#
-## Install packages
-#sudo apt-get update && \
-#sudo apt-get install -y \
-#    openssh-client \
-#    sshpass
-
-
-exec_cmd "TRACE" echo "Generating BPM RFFE IP mapping file..."
-
-# Generate HALCS config from template
-BPM_MAX_NUM_BOARDS=12
-BPM_MAX_NUM_HALCS=2
-cp ${BPM_HALCS_CFG_TEMPLATE_IN_FILE} ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE}
-for board in `seq 1 ${BPM_MAX_NUM_BOARDS}`; do
-    for halcs in `seq 0 $((${BPM_MAX_NUM_HALCS}-1))`; do
-        bpm_rffe_ip_idx="${board}${halcs}"
-        bpm_rffe_ip="BPM_RFFE_IP_MAPPING[${bpm_rffe_ip_idx}]"
-        bpm_rffe_proto="BPM_RFFE_PROTO_MAPPING[${bpm_rffe_ip_idx}]"
-        sed -i \
-            -e "s#<RFFE_BOARD${board}_HALCS${halcs}_IP>#${!bpm_rffe_ip}#" \
-            -e "s#<RFFE_BOARD${board}_HALCS${halcs}_PROTO>#${!bpm_rffe_proto}#" \
-            ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE}
-    done
-done
-
-
-exec_cmd "TRACE" echo "Modifying EPICS PV prefixes..."
-
-# Login via SSH and setup configuration files
-# Set the --static, --transient and --pretty
-# hostnames by the HOSTNAME variable contents
-SSHPASS="${SSHPASS_USR}" sshpass -e \
-    ssh -o StrictHostKeyChecking=no \
-    root@${IP} \
-    bash -c "\
-        echo \"\" > /etc/hostname && \
-        sysctl kernel.hostname=\"${HOSTNAME}\" > /dev/null && \
-        sysctl -w kernel.hostname=\"${HOSTNAME}\" > /dev/null && \
-        hostnamectl set-hostname \"${HOSTNAME}\" > /dev/null && \
-        sed -i -e \"\
-            { \
-                s|EPICS_PV_CRATE_PREFIX=.*\$|EPICS_PV_CRATE_PREFIX=${CRATE}|; \
-            }\" ${BPM_EPICS_CFG_FILE} && \
-        sed -i -e \"\
-            { \
-                s|EPICS_PV_CRATE_PREFIX=.*\$|EPICS_PV_CRATE_PREFIX=${CRATE}|; \
-            }\" ${TIM_RX_EPICS_CFG_FILE} \
-    "
-
-exec_cmd "TRACE" echo "Copying generated files to CPU..."
-
-SSHPASS="${SSHPASS_USR}" sshpass -e \
-    scp \
-    ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE} \
-    root@${IP}:${BPM_HALCS_CFG_FILE}
-
-exec_cmd "INFO " echo "CPU Configured Successfully!"
+if [ "$CRATE_NUM" -eq 21 ]; then
+    CRATE_HOSTNAME="ia-20rabpmtl-co-iocsrv"
+elif [ "$CRATE_NUM" -gt 21 ]; then
+    CRATE_HOSTNAME="de-${CRATE_NUM}rabpm-co-iocsrv"
+else
+    CRATE_HOSTNAME="ia-${CRATE_NUM}rabpm-co-iocsrv"
+fi
+
+sysctl -w kernel.hostname="$CRATE_HOSTNAME"
+hostnamectl hostname "$CRATE_HOSTNAME"
+
+apt-get update -y
+apt-get dist-upgrade -y
+apt-get install build-essential vim tmux git rsync sshfs telnet screen picocom earlyoom procserv arch-install-scripts zabbix-agent2 tcpdump gdb strace wget lsb-release iptables iptables-persistent systemd-timesyncd -y
```

Comment:
Installing `iptables-persistent´ makes it impossible to run this script unattended due a prompt during the installation process, couldn't find a clean way to solve this. @ericonr @henriquesimoes any idea?

Reply:
Have you tested exporting `DEBIAN_FRONTEND=noninteractive`?

We commonly use this in container files to avoid any interactive prompts during installation.

Reply:
No luck...

Reply:
`doas env DEBIAN_FRONTEND=noninteractive dpkg-reconfigure iptables-persistent` seems to have worked for me. It didn't show a prompt when reconfiguring. I don't think there should be a difference when using `apt-get install`, but I'm not sure. How did you test it?